### PR TITLE
Julia for BSSw blog article first draft

### LINKED ISSUE #1470: Possible Blog Article: Julia's value proposition for better scientific software
Title: "The Julia programming language value proposition for better scientific software"

Short abstract:
Julia is a dynamic open-source language designed with scientific computing (Fortran-like) and data science (Python-like) friendly syntax. Julia code builds upon LLVM for performance on CPU and GPU heterogeneous hardware, prioritizes lightweight interoperability with existing C and Fortran frameworks, while providing a unified language ecosystem for data analysis, plotting, interactive computing, reproducibility, testing, instrumentation, and packaging. Julia is a one-stop-shop alternative to the N-language+ecosystem programming components used in the increasingly expensive and complex process of rapid prototyping going from ideas to performance portable and composable simulation+data workflows. In this article we describe the value proposition of Julia for the communities invested in the different angles of better scientific software.

### PR DESCRIPTION
**Pull request by williamfgc**
# Description
Add blog article titled "Julia value proposition for Better Scientific Software"

EB Member: @bernhold

Resolves #1470 

1 markdown text file and 3 figures. First draft any peer-review and feedback is welcome.

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [x] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [ ] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [ ] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [ ] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [ ] ***[EB Mem]*** Add one or more Reviewers.
* [ ] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [ ] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [ ] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [ ] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [ ] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [ ] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [ ] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [ ] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [ ] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [ ] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

## PR checklist for (internal) files not displayed on bssw.io site

*Click "Write" above and remove comment markers to see below checklist items*

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by bernhold** [comment]
Oops, I didn't pay enough attention to where I was posting.  Please ignore the [request on the issue](https://github.com/betterscientificsoftware/bssw.io/issues/1470#issuecomment-1487360579) and answer here instead.

Hi @williamfgc, I'm wondering if you have thoughts about a hero image for your blog post. The guidelines are below. But I was thinking that maybe you had either a Julia-based science image to share, or a "people photo" from one of the recent Julia events that you've organized? We work with a designer who can handle any cropping and other adjustments that might be necessary, so all we really need from you is a candidate image or two. We can also go with a stock image or with no hero at all.

At the top of your article, we can include a "hero" image, which is also used as a background to highlight it as the newest article in the list of blog postings. This image is optional, but if you have ideas for an appropriate image please let us know. Many blog articles are published without hero images. Guidelines for "hero" image:

- wide rectangular shape; we can crop an image to the required size (1125 x 432 pixels)
- the design brief for our site focuses on three types of imagery for hero:
  - stock images (we have access to a number of stock photo libraries),
  - data visualizations, or
  - people in trainings, workshops, conferences, or other settings
- "hero" images should contain no bold text; incidental text (in diagrams, for instance) is OK.

**Comment by williamfgc** [comment]
Hi @bernhold , thanks for the feedback. We can reuse some of the images from the SC22 BoF in this [repo](https://github.com/JuliaParallel/julia-hpc-bof-sc22). In particular the one @JBlaschke did attached here 
JuliaForHPC_BoF_SC22. Let me know what you think.

**Comment by JBlaschke** [comment]
Hi All,

I just added the SVG sources for the original banner here: https://github.com/JuliaParallel/julia-hpc-bof-sc22/tree/main/assets if you have a high-res version of that backround image, then you can overlay our banner with that.

You have my permission to make edits to the svg btw.

**Comment by bernhold** [comment]
@williamfgc and @JBlaschke thanks for your help with images.  While I start the conversation with our designer, I wanted to just double check with for a couple of other types of images we often us (more options is better).
1. People photos, e.g., from your recent BoF or any of the tutorials William has organized, or anything like that
2. A science image that is based somehow on Julia -- maybe a visualization from a simulation code that happens to be in Julia or some such?  It doesn't have to be something that you generated as long as whoever did is willing to let us use it.

Just let me know if you have anything like that.  Thanks

**Comment by williamfgc** [comment]
@bernhold thanks for following up and thanks @JBlaschke for chiming in. If it helps, I have a [photo album](https://photos.app.goo.gl/7qnGbXaR6aXtFEdBA) with people's photos from the Julia BoF at SC. Most meetings have been virtual. On 2. there are too many options.

**Comment by bernhold** [comment]
Thanks for the BoF photos.

> On 2. there are too many options.

I'm not asking for all science images with some Julia connection.  Maybe there are O(2) that you like that we can put into the pool of hero candidates?

File: Articles/Blog/2023-03-Julia_for_BSSw.md [hunk 1]
```diff
@@ -1,52 +1,60 @@
...
+*David: I think this is an unnecessary diversion and would like to delete it.*
+Despite being an amortized cost, the "time to first plot" (TTFP) is a recurrent issue that is actively being addressed in recent versions, see discourse [thread](https://discourse.julialang.org/t/julia-v1-9-0-beta2-is-fast/92290),
```
Comment (williamfgc) [inline Articles/Blog/2023-03-Julia_for_BSSw.md:57]: Agree!

File: Articles/Blog/2023-03-Julia_for_BSSw.md [hunk 2]
```diff
@@ -63,33 +71,37 @@ $ julia
...
+For a more technical introduction to the Julia ecosystem, you might want to start with a blog article that I recently updated: [First Project using the Julia Language](https://williamfgc.github.io/programming/scientific-computing/2023/03/03/first-project-julia-language.html).  It includes links to numerous other resources which can provide further help getting started. I'd recommend to anyone trying the language for the first time to use the [Visual Studio Code](https://code.visualstudio.com/), which offers excellent Julia support through its extensions marketplace, and the upcoming [Julia v1.9 version](https://julialang.org/downloads/#upcoming_release) for a better experience.
+I also encourage you to seek out colleagues who may be using Julia and build a local community, including sharing experiences, tips, and examples.
+Ultimately, scientific software, in Julia or other languages, benefits from communities.
+*Bonus: I've found that using GitHub copilot with the simple Julia APIs can be a welcome boost to the productivity of writing Julia code. Its autocompletion-like capabilities save on typing (but not thinking).*
```
Comment (williamfgc) [inline Articles/Blog/2023-03-Julia_for_BSSw.md:91]: I think the official name is "GitHub Copilot"

Comment (williamfgc) [reply Articles/Blog/2023-03-Julia_for_BSSw.md:91]: just minor: "writing and porting code to Julia"

File: Articles/Blog/2023-03-Julia_for_BSSw.md [hunk 3]
```diff
@@ -1,52 +1,60 @@
...
+Julia enables lightweight reusability of existing Fortran and C infrastructure via the [`@ccall` macro](https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/). Similarly, Python and R interoperability is possible with [PyCall.jl](https://github.com/JuliaPy/PyCall.jl) and [RCall.jl](https://github.com/JuliaInterop/RCall.jl), respectively. So Julia promotes reuse over reinvention, which is both important and useful given the volume of mature scientific software.
+
+**Jupyter and Pluto.jl notebooks**
+Computational notebooks, Jupyter in particular, have been widely adopted in science and do not need an introduction. Jupyter is powered by Anaconda with Python kernels requiring setting a conda environment for managing the required dependencies before launching the server and web client interface. Notebooks are stored using a `*.ipynb` file format based on JSON. 
+
+*David: I don't understand this sentence.  Are we talking about Jupyter still or Pluto?  Should this sentence be in the following paragraph?*
```
Comment (williamfgc) [inline Articles/Blog/2023-03-Julia_for_BSSw.md:51]: Good catch. I would rephrase it as: "Jupyter supports interactive Julia notebooks via the [IJulia](https://github.com/JuliaLang/IJulia.jl) kernel package that can understand information readily available..."

**Review by williamfgc — COMMENTED** [review]
@bernhold thanks for the review. I made some comments hoping to address the changes

**Comment by williamfgc** [comment]
> I'm not asking for all science images with some Julia connection. Maybe there are O(2) that you like that we can put into the pool of hero candidates?

None I can think of... or I should rephrase, nothing really "unique" to Julia that can't be done with other languages and ecosystems :)
